test/helpers/bucket.js

```diff
@@ -0,0 +1,36 @@
+import { GetObjectCommand } from '@aws-sdk/client-s3'
+import pRetry from 'p-retry'
+
+/**
+ * @param {import('@aws-sdk/client-s3').S3Client} client
+ * @param {string} bucketName
+ * @param {string} key
+ */
+export async function getBucketItem (client, bucketName, key) {
+  const cmd = new GetObjectCommand({
+    Bucket: bucketName,
+    Key: key
+  })
+
+  const response = await pRetry(async () => {
+    let r
+    try {
+      r = await client.send(cmd)
+    } catch (err) {
+      console.log('bucket error', err)
+    }
+
+    if (r?.$metadata.httpStatusCode !== 200) {
+      console.log('bucket error', r?.$metadata.httpStatusCode)
+      throw new Error('not found')
+    }
+
+    return r
+  }, {
+    retries: 10,
+    maxTimeout: 1000,
+    minTimeout: 1000
+  })
+  
+  return await response.Body?.transformToByteArray()
```

Comment:
You throw if `httpStatusCode` is not 200, so this should always have a `Body`.

Pick a mode - throw if not found _or_ return `undefined`. Right now this function throws on 404 _and_ (as per the types) may return `undefined`.

Reply:
yes, but again types don't really behave as you expect. Welcome to aws types 😉 https://github.com/aws/aws-sdk-js-v3/blob/932bdfa8e6fecc22f6f4985b338fe3645f4a5b86/clients/client-mediastore-data/src/commands/GetObjectCommand.ts#L37